### PR TITLE
Fixes #33008 - switch to sql schema dumping

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: config/initializers/dynflow.rb [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # Ignore Dynflow tables when schema-dumping. Dynflow tables are handled automatically by Dynflow.
-ActiveRecord::SchemaDumper.ignore_tables << /^dynflow_.*$/
+ActiveRecord::SchemaDumper.ignore_tables << '^dynflow_*'
```
Comment (jlsherrill) [inline config/initializers/dynflow.rb:2]: I'm not 100% sure about this, the docs seem to indicate that `'^dynflow_*$'` should work, but it didn't seem to and allowed everything through. ` '^dynflow_*'` appeared to work fine:
```
$ grep dynflow_ db/structure.sql 
    dynflow_execution_plan_id character varying,
    dynflow_step_id integer,
```

**Review by ekohl — COMMENTED** [review]
@tbrisker thoughts on merging this so it gets in 3.0?

**Review by tbrisker — APPROVED** [review]
Thanks @jlsherrill ! let's get this in to 3.0, hopefully it won't break anything.

**Comment by lzap** [comment]
I am seeing this file as modified all the time, shall we add it into `.gitignore` just like we did with `db/schema.rb`?

File: config/initializers/dynflow.rb (hunk 1 above)
Comment (lzap) [reply config/initializers/dynflow.rb:2]: This actually does NOT work and is a regression. When I do `rake db:schema:dump` it creates me Ruby version (`schema.rb`) and it does contain dynflow tables. This leads to the issue: https://projects.theforeman.org/issues/30052

**Comment by lzap** [comment]
How this is supposed to work? When I do `rake db:schema:dump` on `develop` today, it creates `config/schema.rb`. I don't see any SQL generated.

**Comment by lzap** [comment]
Oh I understand, it creates both `db/schema.rb` AND `db/structure.sql`. Is this expected? What format should rails "load" pick?

**Comment by lzap** [comment]
Looks like Ruby codebase explains it nicely.

```ruby
    ##
    # :singleton-method:
    # A list of tables which should not be dumped to the schema.
    # Acceptable values are strings as well as regexp if ActiveRecord::Base.schema_format == :ruby.
    # Only strings are accepted if ActiveRecord::Base.schema_format == :sql.
    cattr_accessor :ignore_tables, default: []

    ##
    # :singleton-method:
    # Specify a custom regular expression matching foreign keys which name
    # should not be dumped to db/schema.rb.
    cattr_accessor :fk_ignore_pattern, default: /^fk_rails_[0-9a-f]{10}$/

    ##
    # :singleton-method:
    # Specify a custom regular expression matching check constraints which name
    # should not be dumped to db/schema.rb.
    cattr_accessor :chk_ignore_pattern, default: /^chk_rails_[0-9a-f]{10}$/
```

**Comment by lzap** [comment]
I have found out that regular expression works both for SQL and Ruby however regexp in a string only works for SQL. Filing an issue and a PR.

**Pull request merged**